[Linux] Failing to compile on Arch after imgui implementation

using 
`cmake -S . -B build/ -DENABLE_QT_GUI=ON`
`cmake --build . --parallel$(nproc)`

```shell
/shadPS4/src/imgui/renderer/imgui_core.cpp: In function ‘void ImGui::Core::Render(const vk::CommandBuffer&, Vulkan::Frame*)’:
/shadPS4/src/imgui/renderer/imgui_core.cpp:189:5: error: ambiguous overload for ‘operator=’ (operand types are ‘vk::Rect2D’ and ‘<brace-enclosed initializer list>’)
  189 |     };
      |     ^
In file included from /usr/include/vulkan/vulkan.hpp:8579,
                 from /shadPS4/src/video_core/renderer_vulkan/vk_common.h:17,
                 from /shadPS4/src/video_core/renderer_vulkan/vk_platform.h:11,
                 from /shadPS4/src/video_core/renderer_vulkan/vk_instance.h:9,
                 from /shadPS4/src/imgui/renderer/imgui_core.h:6,
                 from /shadPS4/src/imgui/renderer/imgui_core.cpp:9:
/usr/include/vulkan/vulkan_structs.hpp:8142:14: note: candidate: ‘vk::Rect2D& vk::Rect2D::operator=(const VkRect2D&)’
 8142 |     Rect2D & operator=( VkRect2D const & rhs ) VULKAN_HPP_NOEXCEPT
      |              ^~~~~~~~
/usr/include/vulkan/vulkan_structs.hpp:8124:10: note: candidate: ‘constexpr vk::Rect2D& vk::Rect2D::operator=(const vk::Rect2D&)’
 8124 |   struct Rect2D
      |          ^~~~~~
/usr/include/vulkan/vulkan_structs.hpp:8124:10: note: candidate: ‘constexpr vk::Rect2D& vk::Rect2D::operator=(vk::Rect2D&&)’
make[2]: *** [CMakeFiles/shadps4.dir/build.make:289: CMakeFiles/shadps4.dir/src/imgui/renderer/imgui_core.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:247: CMakeFiles/shadps4.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```
i made sure to update my system and install imgui from the AUR just in case

a bit unrelated but it also throws this warning
```shell
In file included from /shadPS4/externals/dear_imgui/imgui.h:64,
                 from /shadPS4/src/imgui/renderer/imgui_core.cpp:5:
/shadPS4/src/imgui/imgui_config.h:28:100: warning: backslash-newline at end of file
   28 | #define IM_VEC2_CLASS_EXTRA
```

--- a/src/imgui/renderer/imgui_core.cpp
+++ b/src/imgui/renderer/imgui_core.cpp
@@ -183,7 +183,7 @@ void Render(const vk::CommandBuffer& cmdbuf, ::Vulkan::Frame* frame) {
         },
     };
     vk::RenderingInfo render_info = {};
-    render_info.renderArea = {
+    render_info.renderArea = vk::Rect2D{
         .offset = {0, 0},
         .extent = {frame->width, frame->height},
     };

